services/staging/rs_server_staging/main.py

```diff
@@ -55,10 +58,55 @@
     allow_headers=["*"],
 )
 
-api = API(get_config(os.environ["PYGEOAPI_CONFIG"]), os.environ["PYGEOAPI_OPENAPI"])
 
+# Exception handlers
+@app.exception_handler(StarletteHTTPException)
+async def custom_http_exception_handler(
+    request: Request,
+    exc: StarletteHTTPException,
+):  # pylint: disable= unused-argument
+    """HTTP handler"""
+    return JSONResponse(status_code=exc.status_code, content={"message": exc.detail})
+
+
+os.environ["PYGEOAPI_OPENAPI"] = ""  # not used
+
+
+def get_config_path() -> pathlib.Path:
+    """Return the pygeoapi configuration path and set the PYGEOAPI_CONFIG env var accordingly."""
+    path = pathlib.Path(__file__).parent.parent / "config" / "staging.yaml"
+    os.environ["PYGEOAPI_CONFIG"] = str(path)
+    return path
 
-def init_db():
+
+def get_config_contents() -> dict:
+    """Return the pygeoapi configuration yaml file contents."""
+    # Open the configuration file
+    with open(get_config_path(), encoding="utf8") as opened:
+        contents = opened.read()
+
+        # Replace env vars by their value
+        contents = os.path.expandvars(contents)
```

Comment:
It is about the personal choice here. I would rather use the previous code in which  the loading of the config file is explicitly stopped at the first not found env var, and an appropriate message is printed as well as stopping the execution of the app. The expandvars will just let any unfound env var with that value, and Postgresql Manager will complain later on about this with a traceback exception. As I said, is about the vision, in my case I would go with the def get_manager_def(): function that replaces the placeholders

Reply:
Is it OK for you if instead of using `os.path.expandvars` we use `(Template('$HOME/stuff').substitute(os.environ)` ? See: https://stackoverflow.com/a/37800939. It will also raise a KeyError at the first unfound env var. And it handles both the `$POSTGRES_HOST` and `${POSTGRES_HOST}` syntax.

Reply:
Ok with this approach.